### PR TITLE
improve handling of dev help topics

### PR DESCRIPTION
**Pull request by kevinushey**
### Intent

Addresses https://github.com/rstudio/rstudio/issues/14273.

### Approach

Check for errors when attempting to render help, and bail if an error occurs.

### Automated Tests

N/A

### QA Notes

Test via notes in https://github.com/rstudio/rstudio/issues/14273.

### Documentation

N/A

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

File: src/cpp/session/modules/SessionHelp.R [hunk 1]
```diff
@@ -795,38 +800,14 @@ options(help_type = "html")
...
       dirpath <- dirname(path)
-      pkgname <- basename(dirpath)
+      package <- basename(dirpath)
       
-      query <- paste("/library/", pkgname, "/html/", basename(file), ".html", sep = "")
+      query <- sprintf("/library/%s/html/%s.html", package, basename(file))
       html <- suppressWarnings(tools:::httpd(query, NULL, NULL))$payload
    }
    
-   # resolve associated package from helpfile path -- note that help file
```
Comment (kevinushey) [inline src/cpp/session/modules/SessionHelp.R:804]: This should be dead code -- we now set `package` in the two branches above, and (conceptually) a help topic should always be associated with a package anyhow.

**Comment by olivroy** [comment]
By loading the new getHelp function, it works. Furthermore, we get help text for arguments, which is great, and did not work previously! So, this PR actually improves UX #13526

image

**Review by gtritchie — APPROVED** [review]
LGTM!

**Comment by gtritchie** [comment]
Add NEWS entry, if needed.

**Pull request merged**